src/Repository/Repository.php

```diff
@@ -488,13 +489,15 @@ final public function ensure(callable $callback, int $retryTimes = 3, bool $reco
         while ($attempts < $retryTimes) {
             try {
                 return $callback($this);
-            } catch (Throwable $e) {
+            } catch (ConnectionException $e) {
                 if ($attempts++ === $retryTimes) {
                     throw $e;
                 }
                 if ($reconnect) {
                     $this->getExplorer()->getConnection()->reconnect();
                 }
+            } catch (Throwable $e) {
+                throw $e;
```

Comment:
he?

Reply:
Mali sme konkretny priklad kedy hermes insertoval s duplicate key a v podstate to padalo iba na ensure a neprichadzala ziadna chybova hlaska. Az kym som si to nevyskusal lokalne bez ensure. Tento kus kodu mam testnuty.

Reply:
lulco myslí že je zbytočne catchovať exceptionu aby si ju znova vyhodil, stačí tie riadky dať preč a stane sa to isté.

mne to ušlo.

Reply:
Jasné.. taký nervný som bol z hľadania, kde je chyba :D .. upravené: https://github.com/efabrica-team/nette-database-repository/pull/12